src/first_pass.c

```diff
@@ -324,169 +362,289 @@ boolean operation(char* asm_line)
 	{
-		code_table[I].c.role = 4; /*Its absolute*/
-		code_table[I].c.address = IC; /*Give address*/
-		IC++;
-		code_table[I].c.op_code = opcode;
-		code_table[I].c.next = NULL;
+		code_table[I].c.source_indirect_register = 1;
+		code_table[I].c.destination_indirect_register = 1;
 	}
 	else
 	{
-		/*Allocate two more memmory words*/
-		temp = (machine_word*)malloc(sizeof(machine_word));
-		if (temp == NULL)
-		{
-			error_log("Memory allocation failure");
-			return FALSE;
-		}
-		temp->c.next = NULL;
-		code_table[I].c.next = temp;
-		temp = (machine_word*)malloc(sizeof(machine_word));
-		if (temp == NULL)
-		{
-			error_log("Memory allocation failure");
-			return FALSE;
-		}
-		temp->c.next = code_table[I].c.next;
-		code_table[I].c.next = temp;
-		code_table[I].c.role = 4; /*Its absolute*/
-		code_table[I].c.address = IC; /*Give address*/
-		IC++;
-		code_table[I].c.next->c.address = IC;
-		IC++;
-		code_table[I].c.next->c.next->c.address = IC;
-		IC++;
-		code_table[I].c.op_code = opcode;
-		/*For destination*/
-		if (operand_destination == '0')
-			code_table[I].c.destination_immidiate = 1;
-		else if (operand_destination == '1')
-			code_table[I].c.destination_direct = 1;
-		else if (operand_destination == '2')
-			code_table[I].c.destination_indirect_register = 1;
-		else
-			code_table[I].c.destination_direct_register = 1;
-		/*For source*/
-		if (operand_source == '0')
-			code_table[I].c.source_immidiate = 1;
-		else if (operand_source == '1')
-			code_table[I].c.source_direct = 1;
-		else if (operand_source == '2')
-			code_table[I].c.source_indirect_register = 1;
-		else
-			code_table[I].c.source_direct_register = 1;
+		code_table[I].c.source_direct_register = 1;
+		code_table[I].c.destination_direct_register = 1;
 	}
+	return TRUE;
+}
+
+/**
+ * @brief Configures the machine word for the destination operand in the code table.
+ *
+ * This function sets up a `machine_word` structure for the destination operand by linking it into the existing code table.
+ * It assigns the role and address fields, sets the operation code, and configures the appropriate addressing mode
+ * based on the provided destination operand. The function also ensures that the instruction counter (IC) is incremented
+ * as addresses are assigned.
+ *
+ * @param temp A pointer to the `machine_word` structure that is being configured.
+ * @param operand_dst The addressing mode of the destination operand. Valid values are:
+ *        - '0': Immediate addressing mode
+ *        - '1': Direct addressing mode
+ *        - '2': Indirect register addressing mode
+ *        - Any other value is treated as direct register addressing mode.
+ * @return A boolean value indicating success (TRUE). Currently, the function always returns TRUE.
+ */
+boolean configure_destination_operand(machine_word* temp, char operand_dst) {
+	temp->c.next = NULL;
+	code_table[I].c.next = temp;
+	code_table[I].c.role = 4; /*Its absolute*/
+	code_table[I].c.address = IC; /*Give address*/
+	IC++;
```

Comment:
DO NOT USE GLOBALS
![image](https://github.com/user-attachments/assets/67d461de-f4d5-4916-8927-a3d526ea1bfb)